tools/codegen/core/gen_upb_api.sh

```diff
@@ -124,3 +124,18 @@ do
     --upb_out=$UPB_OUTPUT_DIR \
     --plugin=protoc-gen-upb=$UPB_PLUGIN
 done
+
+# In PHP build Makefile, the files with .upb.c suffix collide .upbdefs.c suffix due to a PHP buildsystem bug.
+# Work around this by placing the generated files with ".upbdefs.c" suffix under a different directory.
+# See https://github.com/grpc/grpc/issues/23307
+
+# move all .upbdefs.c files from under src/core/ext/upb-generated to src/core/ext/upbdefs-generated
+UPBDEFS_OUTPUT_DIR=$PWD/src/core/ext/upbdefs-generated
+rm -rf $UPBDEFS_OUTPUT_DIR
+cp -r $UPB_OUTPUT_DIR $UPBDEFS_OUTPUT_DIR
```

Comment:
Is it safe to not have a `mkdir -p $UPBDEFS_OUTPUT_DIR` prior to this?

Reply:
Yes, it works. The cp command takes the entire upb-generated and stores a copy as upbdefs-generated (if the destination directory would exist, the outcome would be different (upb-generated would end up under upbdefs-generated/upb-generated), but we do `rm -rf $UPBDEFS_OUTPUT_DIR` in the previous step.